data_steward/validation/main.py

```diff
@@ -286,36 +286,54 @@ def process_hpo(hpo_id, force_run=False):
             logging.info('Uploading achilles index files to `gs://%s/%s`.', bucket, folder_prefix)
             _upload_achilles_files(hpo_id, folder_prefix)
 
-        # get top heel errors
-        heel_result = get_heel_result(hpo_id)
-
-        heel_errors = _convert_heel_result_to_csv(heel_result)
+        # Get heel errors
+        heel_errors, heel_header_list = get_heel_errors_in_results_html(hpo_id,
+                                                                        common_sql.HEEL_ERROR_QUERY_VALIDATION,
+                                                                        common.ACHILLES_HEEL_RESULTS_VALIDATION)
 
-        heel_error_header = get_heel_error_header(heel_result)
+        # Get Drug check counts into results.html
+        drug_checks, drug_header_list = get_drug_checks_in_results_html(hpo_id,
+                                                                        common_sql.DRUG_CHECKS_QUERY_VALIDATION,
+                                                                        common.DRUG_CHECK_TABLE_VALIDATION)
 
         _save_results_html_in_gcs(hpo_id, bucket, folder_prefix + common.RESULTS_HTML, results, errors, warnings,
-                                  heel_errors, heel_error_header)
+                                  heel_errors, heel_header_list, drug_checks, drug_header_list)
 
         now_datetime_string = datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S')
         logging.info('Processing complete. Saving timestamp %s to `gs://%s/%s`.',
                      bucket, now_datetime_string, folder_prefix + common.PROCESSED_TXT)
         _write_string_to_file(bucket, folder_prefix + common.PROCESSED_TXT, now_datetime_string)
 
 
-def _convert_heel_result_to_csv(list_of_dicts):
-    result_list = list()
-    if list_of_dicts is []:
-        return result_list.append(tuple())
+def get_heel_errors_in_results_html(hpo_id, query_string, table_id):
+    heel_result = get_query_result(hpo_id, query_string, table_id)
+    heel_errors = _convert_query_result_to_list_of_rows(heel_result)
+    if not heel_result:
+        heel_header_list = ['Record Count', 'Heel Error', 'Analysis ID', 'Rule ID']
     else:
-        for dict_item in list_of_dicts:
-            dict_values = tuple(dict_item.values())
-            result_list.append(dict_values)
-        return result_list
+        heel_header_list = _get_query_result_header(heel_result)
+    return heel_errors, heel_header_list
+
```

Comment:
Both `get_heel_errors_in_results_html()` and `get_heel_errors_in_results_html` perform the same function except for the headers. They can be generalized to `add_table_in_results_html()` which takes in an additional input for headers.